### PR TITLE
Add support to HWP in beam convolution

### PR DESCRIPTION
**Pull request by keskitalo**

### CODE REVIEW

**Comment by keskitalo** [comment]
Fix the divergent history of #534

File: cmake.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX
```
Comment (tskisner) [inline cmake.sh:1]: Was this script accidentally committed?  If you have a specific set of cmake options needed for a system, feel free to add a script to the `platforms` directory.  We could consider adding a `default.sh` script or similar.

File: src/toast/ops/conviqt.py [hunk 2]
```diff
@@ -266,6 +266,8 @@ def _exec(self, data, detectors=None, **kwargs):
             detector = self.get_detector(det)
 
             theta, phi, psi, psi_pol = self.get_pointing(data, det, verbose)
+            np.savez(f'{self.det_data}_{det}.npz', psi=psi, psi_pol=psi_pol,
+                    theta=theta, phi=phi)
```
Comment (tskisner) [inline src/toast/ops/conviqt.py:270]: Could we guard this debug dumping by a line like:
```
# (assuming that Environment was imported at the top of the file)
env = Environment.get()
if env.log_level == "VERBOSE":
    np.savez(...)
```
or use some other internal `verbose` settting in the class.

File: src/toast/tests/ops_sim_tod_conviqt.py [hunk 3]
```diff
@@ -96,19 +128,111 @@ def setUp(self):
...
+                self.beam_file_dict[det] = fname2
+
+        return
+
+    def test_sim_conviqt(self):
         if not ops.conviqt.available():
             print("libconviqt not available, skipping tests")
             return
 
         # Create a fake scan strategy that hits every pixel once.
```
Comment (tskisner) [inline src/toast/tests/ops_sim_tod_conviqt.py:230]: Remove stale comment

File: src/toast/tests/ops_sim_tod_conviqt.py [hunk 4]
```diff
@@ -202,99 +320,263 @@ def test_sim(self):
...
-            norm = np.mean(cl_out[compare] / ref)
-            if not np.allclose(
-                norm * ref,
-                cl_out[compare],
-                rtol=1e-5,
-                atol=1e-5,
-            ):
-                fail = True
+        # Create a fake scan strategy that hits every pixel once.
+        #        data = create_healpix_ring_satellite(self.comm, nside=self.nside)
```
Comment (tskisner) [inline src/toast/tests/ops_sim_tod_conviqt.py:344]: You can remove this, and the comment

**Review by tskisner — CHANGES_REQUESTED** [review]
Just a couple of small tweaks, but otherwise looks good

File: cmake.sh (hunk 1 above)
Comment (keskitalo) [reply cmake.sh]: Oops! Removed the unnecessary file.

File: src/toast/ops/conviqt.py (hunk 2 above)
Comment (keskitalo) [reply src/toast/ops/conviqt.py]: That is a leftover debugging statement. Now removed.

File: src/toast/tests/ops_sim_tod_conviqt.py (hunk 4 above)
Comment (keskitalo) [reply src/toast/tests/ops_sim_tod_conviqt.py]: Fixed.

**Comment by keskitalo** [comment]
Addressed review comments and will merge once the unit tests finish.

**Pull request merged**